### ISSUE TITLE
High fluctuation in training loss during SFT on RoboTwin tasks - Is this expected?

### ISSUE THREAD

**Issue by lwbscu** [Robbyant/lingbot-vla#21]
Hi LingBot-VLA team,

Thanks for releasing this amazing work and the highly efficient training codebase! 

I am currently running the Post-Training (SFT) pipeline using the provided `lingbot-vla-4b` model and `Qwen2.5-VL-3B-Instruct` base. I have conducted rigorous experiments on both a single-task dataset (e.g., `click_bell`) and the RoboTwin 5-tasks mixed dataset.

### 1. My Training Setup
- **Hardware:** 8x A800 GPUs
- **Batch Size:** `micro_batch_size = 4` per GPU (resulting in `global_batch_size = 32`)
- **Config:** Modified from `robotwin_load20000h.yaml`

### 2. Observation A: High Loss Fluctuation
During training, I plotted the loss from `loss.jsonl`. Interestingly, **even in the single-task training scenario** (as shown in the attached figure for the `click_bell` task), the raw training loss (blue line) fluctuates heavily with dense spikes. 

<img width="2545" height="1651" alt="Image" src="https://github.com/user-attachments/assets/5966dc03-2b22-4652-919b-f08e512d2ebb" />

### 3. Observation B: Significant Evaluation Performance Gap
While the official claims demonstrate a highly capable model (e.g., ~80%+ success rates on RoboTwin tasks), my evaluation results fall far short of this, despite strictly following the SFT pipeline:
* **Single-Task SFT:** When training exclusively on the simplest task (`click_bell`), the success rate is only around **30%**.
* **Multi-Task SFT (5 Tasks):** I then strictly followed the official documentation snippet: *"Training Configuration: We provide the mixed post-training configuration in five RoboTwin 2.0 tasks ('open microwave' 'click bell' 'stack blocks three' 'place shoe' 'put object cabinet')."* Surprisingly, when training on the 5-task mixed dataset, the success rate for individual tasks drops drastically to **single digits (just a few percent)**.

<img width="2488" height="2133" alt="Image" src="https://github.com/user-attachments/assets/386662a9-9280-4da5-a36a-09dd1b43ff2b" />

### 4. My Questions:
1. **Loss Fluctuation:** Given the 8-GPU setup (`micro_batch_size=4`), is this high-frequency fluctuation completely normal for LingBot-VLA's architecture during SFT, even on a single task? How do you usually determine optimal convergence?
2. **Performance Reproduction:** How can we reproduce the high success rates mentioned in your reports? Are there specific hyperparameters, data augmentation strategies, or longer training epochs required that are not reflected in the default `robotwin_load20000h.yaml`?
3. **Multi-task Degradation:** Is it expected that the 5-task mixed training performs significantly worse than single-task training under the current config? Does the multi-task setup require a different sampling strategy or a much larger batch size to handle task interference?

Thanks in advance for your time and insights!

**Comment by zxduan-r** [comment]
Thanks for the official clarification! We'd like to share our reproduction results as additional reference for the community.

## Our Setup

We also trained on the **5-task mixed dataset** using the `robotwin_load20000h.yaml` config, but with a larger global batch size via multi-node training:

| Parameter | Value |
|-----------|-------|
| Hardware | 2 nodes × 8 GPUs (16 GPUs total) |
| `micro_batch_size` | 32 |
| `global_batch_size` | 512 |
| `lr` | 1e-4 |
| `num_train_epochs` | 69 |
| Effective steps | 10,005 |
| `loss_type` | L1_fm |
| Base model | `lingbot-vla-4b` |

Training loss converged from ~0.53 → ~0.011.

## Evaluation Results (unseen instructions, 100 trials, `demo_clean`)

| Task | Success Rate |
|------|-------------|
| open_microwave | 75/100 = **75.0%** |
| click_bell | 66/100 = **66.0%** |
| place_shoe | 48/100 = **48.0%** |
| stack_blocks_three | 26/100 = **26.0%** |
| put_object_cabinet | 22/100 = **22.0%** |
| **Average** | **47.4%** |

Compared to the results reported in issue #21 (single-digit % with GBS=32), increasing the global batch size to 512 with multi-node training seems to significantly improve performance. This is consistent with the team's suggestion to increase batch size and training steps.

That said, there is still a noticeable gap from the paper's reported ~80%+. We look forward to the release of the fine-tuned weights and the exact training configuration used in the paper. 

Also tagging the related discussion in #24 for cross-reference.
